src/plugins/cjs-interop.ts

```diff
@@ -10,17 +15,81 @@ export const cjsInterop = (): Plugin => {
         this.format !== 'cjs' ||
         info.type !== 'chunk' ||
         !/\.(js|cjs)$/.test(info.path) ||
-        !info.entryPoint ||
-        info.exports?.length !== 1 ||
-        info.exports[0] !== 'default'
+        !info.entryPoint
       ) {
         return
       }
 
+      if (this.splitting) {
+        // there is exports metadata when cjs+splitting is set
+        if (info.exports?.length !== 1 || info.exports[0] !== 'default') return
+      } else {
+        const swc: typeof import('@swc/core') = localRequire('@swc/core')
+        const { Visitor }: typeof import('@swc/core/Visitor') =
+          localRequire('@swc/core/Visitor')
+        if (!swc || !Visitor) {
+          throw new PrettyError(
+            `@swc/core is required for cjsInterop when splitting is not enabled. Please install it with \`npm install @swc/core -D\``
+          )
+        }
+
+        let entrySource: string | undefined
```

Comment:
Why don't use `code` param directly?

Reply:
`code` has already been transpiled to cjs, but we need the source code here

Reply:
This could break the plugin system. If a plugin attaches a named export, then cjs interop cannot analyze it.

Reply:
you're right, but cjs is difficult to analyze exports, and esbuild does not provide output meta either when the format is cjs

Reply:
Hmmm can we just enable spiltting automatically if `cjsInterop` is enable (add the description to the docs)